be/src/fs/fs_starlet.cpp

```diff
@@ -162,65 +210,71 @@ class StarletFileSystem : public FileSystem {
 
     StatusOr<std::unique_ptr<WritableFile>> new_writable_file(const WritableFileOptions& opts,
                                                               const std::string& path) override {
-        if (!path.empty() && path.back() == '/') {
+        ASSIGN_OR_RETURN(auto pair, parse_starlet_path(path));
+        if (!pair.first.empty() && pair.first.back() == '/') {
             return Status::NotSupported(fmt::format("Starlet: cannot create file with name ended with '/': {}", path));
         }
-        auto format_str = parse_starlet_path(path);
-        DCHECK(g_starlet != nullptr);
-        staros::starlet::ObjectStorePtr object_store = g_starlet->get_store(format_str);
-        if (object_store == nullptr) {
-            return Status::InternalError(fmt::format("Failed to get store from starlet path {}", path));
+
+        ASSIGN_OR_RETURN(auto conf, get_shard_config(pair.second))
+
+        auto fs_st = FileSystemFactory::new_filesystem("s3://", conf);
```

Comment:
better the result of `new_filesystem()` should be cached in StarletFileSystem, otherwise fs_st will be destroyed at the end and its client factory will be destroyed as well.